### DISCUSSION TITLE
reframe coordinates to be barycentric

### DISCUSSION THREAD

**Discussion by SudatKhan** [PrincetonUniversity/athena#604] [Q&A] [answered]
I've been working on some planet-disk interaction simulations where you have a planet embedded within a disk orbiting a central star. In the disk.cpp file, GM in the input file is treated as a point mass for the star at the center of the coordinates. I've been trying to run the simulation for the case where the star and planet both orbit the barycenter. For this I removed the GM parameter and treated the star as a source term in a binary with the planet, however this hasn't been working as expected. Any ideas of how to move forward?

**Comment by tomidakn** [comment]
I'm afraid I cannot provide any meaningful support without more specific information. What is exactly your problem and what do you expect?

In your coordinate system, the orbit of the gas near the coordinate center is not circular. If you are using spherical polar coordinates, probably you need some special treatment for the inner boundary conditions.

**Reply by SudatKhan** [reply]
Apologies. So essentially my problem setup is using the disk.cpp problem generator that I had modified to include user boundary conditions and viscosity. We have two Jupiter massed planets at fixed circular radii orbiting around a central Sun-massed star. Now according to the documentation for the input file, the parameter 'GM' is used a central point mass. However, I want to be able to have the whole system (star + the two planets) orbit their barycenter. I was wondering if there was a way to delete the central point mass and use a source term to create a new star at a slightly displaced position so that it (along with the planets) would orbit a massless barycenter.

**Comment by tomidakn** [comment]
You can simply delete the 'GM' line from the input file to turn off the point source gravity.
And you need to write your own source function. As the user-defined source function can take time as a parameter, you can calculate the positions of star and planets, and derive the potential/force. Please see [Problem Generators](https://github.com/PrincetonUniversity/athena/wiki/Problem-Generators) in the Wiki for details.

**Comment by c-white** [comment, marked as answer]
Adding to the above, your own source function can imitate the built-in mass at the origin in `src/hydro/srcterms/pointmass.cpp`, where the geometric factors are defined in the appropriate `src/coordinates/` file. The use of face fluxes of mass for modifying the energy is there for good reason, but a quick and dirty cell-centered version using just the primitives can work decently. The only catch is if you decide to go into a rotating frame, since the obvious way of implementing the Coriolis force is wrong, and will cause your system to spiral out of control numerically.
